### PR TITLE
[Modernize Code] Utilize new generic SimpleFactory

### PR DESCRIPTION
**Pull request by azoitl**
Based on the PR https://github.com/eclipse/gef-classic/pull/275 from @ptziegler all usages of SimpleFactory have been rechecked and if possible the typed version is used.

### CODE REVIEW

File: org.eclipse.gef.examples.shapes/src/org/eclipse/gef/examples/shapes/ShapesEditor.java [hunk 1]
```diff
@@ -254,8 +254,9 @@ public void execute(final IProgressMonitor monitor) {
```
Comment (ptziegler) [inline org.eclipse.gef.examples.shapes/src/org/eclipse/gef/examples/shapes/ShapesEditor.java:254]: Sadly, GitHub doesn't let me add comments to unmodified code... But in line 166, SimpleFactory is still a raw type. Given that we work on objects there, I'd suggest something like:
```
return new SimpleFactory<>((Class<?>) template);
```

File: org.eclipse.gef/src/org/eclipse/gef/dnd/TemplateTransferDropTargetListener.java [hunk 2]
```diff
@@ -73,13 +73,15 @@ protected final CreateRequest getCreateRequest() {
...
-			return ((CreationFactory) template);
-		} else if (template instanceof Class) {
-			return new SimpleFactory((Class) template);
-		} else
-			return null;
+		if (template instanceof CreationFactory creationFactory) {
+			return creationFactory;
+		}
+		if (template instanceof Class clazz) {
+			return new SimpleFactory(clazz);
```
Comment (ptziegler) [inline org.eclipse.gef/src/org/eclipse/gef/dnd/TemplateTransferDropTargetListener.java:82]: Similar to the case in ShapeEditor, I'd use something like this to resolve the warning:
```
if (template instanceof Class<?> clazz) {
	return new SimpleFactory<>(clazz);
}
```

**Review by ptziegler — CHANGES_REQUESTED** [review]

**Comment by ptziegler** [comment]
My apologies for not checking the compiler warnings :sweat_smile: I completely forgot to check whether GEF is using them internally...

File: org.eclipse.gef.examples.shapes/src/org/eclipse/gef/examples/shapes/ShapesEditor.java (hunk 1 above)
Comment (azoitl) [reply org.eclipse.gef.examples.shapes/src/org/eclipse/gef/examples/shapes/ShapesEditor.java:254]: I was not sure if Ii would do it. So thx for the feedback fixed it.

File: org.eclipse.gef/src/org/eclipse/gef/dnd/TemplateTransferDropTargetListener.java (hunk 2 above)
Comment (azoitl) [reply org.eclipse.gef/src/org/eclipse/gef/dnd/TemplateTransferDropTargetListener.java]: Done

**Comment by azoitl** [comment]
> My apologies for not checking the compiler warnings 😅 I completely forgot to check whether GEF is using them internally...

No Problem at all. Was a very nice exercise for me to experiment with your change.

**Review by ptziegler — APPROVED** [review]

**Pull request merged**